src/vs/workbench/services/untitled/common/untitledTextEditorModel.ts

```diff
@@ -178,11 +187,23 @@ export class UntitledTextEditorModel extends BaseTextEditorModel implements IUnt
 	private _hasModeSetExplicitly: boolean = false;
 	get hasModeSetExplicitly(): boolean { return this._hasModeSetExplicitly; }
 
-	override setMode(mode: string, setExplicitly = true): void {
+	override setMode(mode: string): void {
+		if (!this._hasModeSetExplicitly
+			&& this.preferredMode
+			&& this.preferredMode !== this.initialMode
+			&& this.languageDetectionService.isEnabledForMode(this.preferredMode)
+		) {
+			// This is a best attempt to narrow it down to: "automatic language detection was wrong"
+			this.telemetryService.publicLog2('automaticlanguagedetection.likelywrong');
```

Comment:
I am not entirely sure this does something useful? If we are trying to detect the case of the auto detection mechanism doing something wrong, how is that related to the initial mode? Shouldn't we rather maybe track the user explicitly changing the mode from the status bar as a indication?

Reply:
Makes sense what Ben is saying. Also feel free to leave this telemetry work as a follow up.